src/features/auth/hocs/withAuthRoute.tsx

```diff
@@ -0,0 +1,34 @@
+import type { NextPage } from 'next'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+import { Routes } from '~/features/core/constants/routes'
+
+import type { UserType } from '../contexts/userContext'
+import { getPersistedUser } from '../storage'
+
+export const withAuthRoute = (WrappedComponent: NextPage): NextPage => {
+  const HOCComponent: NextPage = ({ ...props }) => {
+    const router = useRouter()
+
+    const [user, setUser] = useState<UserType | null>(null)
+
+    useEffect(() => {
+      setUser(getPersistedUser())
```

Comment:
Why not rely on the exiting user context? This line seems not only duplication, but responsibility leak.